Btc-transaction-solidity-helper automated update of documentation (#211)
rsksmart/devportal: rif/rns/integrate/index.md

---
layout: rsk
title: RNS Integration guidelines
tags: rif rns integrate
description: Learn how to integrate RNS with your wallet or dApp.
tags: rns, javascript, dapp, wallet, integrate
---

RIF Name Service (RNS) is a decentralized service that allows users to have a readable domain in any blockchain. It can be used to identify other personal resources, such as payment, ID, storage or communication addresses.

In other words, RIF Name Service (RNS) is the phonebook of the blockchain. Humans access information online through domain names, like alice.rsk or bob.rsk. Wallets interact through the blockchain protocol using addresses. RNS translates domain names to addresses so users can send funds and interact with dapps and people using this domain names.

In order to accomplish that, you should integrate this service to your dApp or wallet. Add the RNS Javascript package to your project to make this process easy.

Just to provide a simple example, let's try to send funds from one wallet to another. So, instead of asking the user to write an unfriendly blockchain address, let's allow him/her to enter just a domain.

> !![](/assets/img/rns/rns-integrate.png)

Reviewer: ```suggestion
![](/assets/img/rns/rns-integrate.png)
```

> title: RNS Integration guidelines

Reviewer: add tags and description

Reviewer 2: ```suggestion
title: RNS Integration guidelines
tags: rif rns integrate
description: Learn how to integrate RNS with your wallet or dApp.
```

---

RIF Name Service (RNS) is a decentralized service that allows users to have a readable domain in any blockchain. It can be used to identify other personal resources, such as payment, ID, storage or communication addresses.

In other words, RIF Name Service (RNS) is the phonebook of the blockchain. Humans access information online through domain names, like alice.rsk or bob.rsk. Wallets interact through the blockchain protocol using addresses. RNS translates domain names to addresses so users can send funds and interact with dapps and people using this domain names.

> In order to accomplish that, you should integrate this service to your dApp or wallet. Is really simple! Just need to add our RNS Javascript package to your project and let magic occurs.

Reviewer: ```suggestion
In order to accomplish that, you should integrate this service to your dApp or wallet. Is really simple! Just need to add our RNS Javascript package to your project and let the magic happen.
```

^ correction for grammar only.

Reviewer: ```suggestion
In order to accomplish that, you should integrate this service to your dApp or wallet. Add the RNS Javascript package to your project to make this process easy.
```

^ alternative without mentioning "magic".

> Just to provide a simple example, let's try to send funds from one wallet to another. So, instead of asking the user to write a non friendly blockchain address, let's allow him/her to enter just a domain.

Reviewer: ```suggestion
Just to provide a simple example, let's try to send funds from one wallet to another. So, instead of asking the user to write an unfriendly blockchain address, let's allow him/her to enter just a domain.
```

> ![](https://i.imgur.com/eO4FG1s.png)

Reviewer: Commit the image into this repo, instead of an external image

RNS will be in charge of translating that domain to an actual blockchain address.

```javascript
const transferToDomain = async (domain) => {
    const address = await rns.addr(domain);

    return transferTo(address);
}
```

> We are pretty sure that the provided example is not enough for all your needings, so let's go deeper on integration.

Reviewer: ```suggestion
We are pretty sure that the provided example is not enough for all your needs, so let's go deeper into integration.
```

  </div>
  {% include rns-integrate-dapp-wallet.html %}
</div>

> _This guides explain how to integrate with RNS within your JavaScript app._

Reviewer: ```suggestion
_This guide explains how to integrate with RNS within your JavaScript app._
```
